File: day2/day2.kt
import java.io.File

fun main() {
    data class Line(val min: Int, val max: Int, val letter: Char, val value: String)
    val input = File("day2.txt").readLines()
        .map { Regex("""(\d+)-(\d+) (\w): (\w+)""").matchEntire(it)!!.groupValues }
        .map { Line(it[1].toInt(), it[2].toInt(), it[3][0], it[4]) }
    println(input.count { x -> x.value.count { it == x.letter } in x.min..x.max })
    println(input.count { x -> listOf(x.min, x.max).count { x.value[it - 1] == x.letter } == 1 })
}

File: 2020/day2/day2.py
import re

input_ = [re.match(r'(\d+)-(\d+) (\w): (\w+)', line).groups()
          for line in open("day2.txt")]
print(sum(1 for a, b, c, s in input_ if int(a) <= len(re.findall(c, s)) <= int(b)))
print(sum(1 for a, b, c, s in input_ if (s[int(a)-1] == c) + (s[int(b)-1] == c) == 1))
